pkg/runtime/runtime.go

```diff
@@ -2135,13 +2136,24 @@ func (a *DaprRuntime) BulkPublish(req *pubsub.BulkPublishRequest) (pubsub.BulkPu
 		return pubsub.BulkPublishResponse{}, runtimePubsub.NotFoundError{PubsubName: req.PubsubName}
 	}
 
-	if allowed := a.isPubSubOperationAllowed(req.PubsubName, req.Topic, ps.ScopedPublishings); !allowed {
+	allowed := a.isPubSubOperationAllowed(req.PubsubName, req.Topic, ps.ScopedPublishings)
+	if !allowed {
 		return pubsub.BulkPublishResponse{}, runtimePubsub.NotAllowedError{Topic: req.Topic, ID: a.runtimeConfig.ID}
 	}
+
 	policyDef := a.resiliency.ComponentOutboundPolicy(req.PubsubName, resiliency.Pubsub)
-	if bulkPublisher, ok := ps.Component.(pubsub.BulkPublisher); ok {
+
+	bulkPublisher, ok := ps.Component.(pubsub.BulkPublisher)
```

Comment:
This should not be here. Instead, use the Features() method prior to this to determine at init time if bulk pubsub is supported natively.

Reply:
ready thanks